core/base/v7/inc/ROOT/RFloat16.hxx

```diff
@@ -0,0 +1,152 @@
+// @(#)root/base
+
+/*************************************************************************
+ * Copyright (C) 1995-2015, Rene Brun and Fons Rademakers.               *
+ * All rights reserved.                                                  *
+ *                                                                       *
+ * For the licensing terms see $ROOTSYS/LICENSE.                         *
+ * For the list of contributors see $ROOTSYS/README/CREDITS.             *
+ *************************************************************************/
+
+#include <cstdint>
+#include <cstring>
+
+#ifndef ROOT_RFloat16
+#define ROOT_RFloat16
+
+/**
+ * Conversion functions between full- and half-precision floats. The code used here is taken (with some modifications)
+ * from the `half` C++ library (https://half.sourceforge.net/index.html), distributed under the MIT license.
+ *
+ * Original license:
+ *
+ * The MIT License
+ *
+ * Copyright (c) 2012-2021 Christian Rau
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+#ifndef HALF_ENABLE_F16C_INTRINSICS
+/// Enable F16C intruction set intrinsics.
+/// Defining this to 1 enables the use of [F16C compiler intrinsics](https://en.wikipedia.org/wiki/F16C) for converting
+/// between half-precision and single-precision values which may result in improved performance. This will not perform
+/// additional checks for support of the F16C instruction set, so an appropriate target platform is required when
+/// enabling this feature.
+///
+/// Unless predefined it will be enabled automatically when the `__F16C__` symbol is defined, which some compilers do on
+/// supporting platforms.
+#define HALF_ENABLE_F16C_INTRINSICS __F16C__
+#endif
+#if HALF_ENABLE_F16C_INTRINSICS
+#include <immintrin.h>
```

Comment:
Just curious, does this actually enable intrinsics on a "standard platform"?

Reply:
Not by default. ROOT would have to be compiled with `-mf16c`, but this doesn't seem to play nice with `projectroot.test.test_TFormulaTests` when added to the CMake configuration. It might be worth it to further investigate and come up with a solution, though.